### ISSUE TITLE
Error running simple execute query

### ISSUE THREAD

**Issue by csga5000** [tekartik/sqflite#46]
When I try to create a database on android, with the following on create:
```
onCreate: (Database db, int version) async {
    await db.execute("create table config (name varchar(32), value varchar(32));");
}
```
The app crashes with this error:

```
E/AndroidRuntime(11343): FATAL EXCEPTION: Sqflite
E/AndroidRuntime(11343): Process: com.example.protectionpro, PID: 11343
E/AndroidRuntime(11343): java.lang.IllegalArgumentException: Cannot bind argument at index 1 because the index is out of range.  The statement has 0 parameters.
E/AndroidRuntime(11343): 	at android.database.sqlite.SQLiteProgram.bind(SQLiteProgram.java:212)
E/AndroidRuntime(11343): 	at android.database.sqlite.SQLiteProgram.bindString(SQLiteProgram.java:166)
E/AndroidRuntime(11343): 	at android.database.sqlite.SQLiteProgram.bindAllArgsAsStrings(SQLiteProgram.java:200)
E/AndroidRuntime(11343): 	at android.database.sqlite.SQLiteDirectCursorDriver.query(SQLiteDirectCursorDriver.java:49)
E/AndroidRuntime(11343): 	at android.database.sqlite.SQLiteDatabase.rawQueryWithFactory(SQLiteDatabase.java:1392)
E/AndroidRuntime(11343): 	at android.database.sqlite.SQLiteDatabase.rawQuery(SQLiteDatabase.java:1331)
E/AndroidRuntime(11343): 	at com.tekartik.sqflite.SqflitePlugin.query(SqflitePlugin.java:511)
E/AndroidRuntime(11343): 	at com.tekartik.sqflite.SqflitePlugin.access$200(SqflitePlugin.java:52)
E/AndroidRuntime(11343): 	at com.tekartik.sqflite.SqflitePlugin$1.run(SqflitePlugin.java:357)
E/AndroidRuntime(11343): 	at android.os.Handler.handleCallback(Handler.java:790)
E/AndroidRuntime(11343): 	at android.os.Handler.dispatchMessage(Handler.java:99)
E/AndroidRuntime(11343): 	at android.os.Looper.loop(Looper.java:164)
E/AndroidRuntime(11343): 	at android.os.HandlerThread.run(HandlerThread.java:65)
I/Process (11343): Sending signal. PID: 11343 SIG: 9
Lost connection to device.
```

**Comment by alextekartik** [comment]
It seems to crash in a query (i.e. not in an execute command). The reports state that you don't have the same number of arguments than the number of `?` (without quotes) in your query so you should verify your queries (see [here](https://stackoverflow.com/questions/20777533/sqlite-cannot-bind-argument-at-index-1-because-the-index-is-out-of-range-the-s) for a similar issue). You can try to run before opening the database:

     Sqflite.devSetDebugModeOn(true);

to get more logs about the offending queries

**Comment by csga5000** [comment]
I saw the error.  And obviously I tried looking at my query.  But in case you didn't see my snippet I have no ?'s in my query.  And am not specifying any args.  Additionally that query runs fine on iOS.  I can't think how this could be anything but a bug in the Android code for this plug-in.  I'll check again when I get to work, to see if I can't see more.

**Comment by csga5000** [comment]
I was breakpointing and stepping over the code, so I felt fairly certain it was in the create, but I suppose it could be a race case or something.  I'll look more

**Comment by csga5000** [comment]
I didn't quite compute what you were saying about execute vs query, but I'm following now.

**Comment by csga5000** [comment]
I've changed the code dramatically since I've continued work on iOS, now that I've pulled android works now until a later point.  But I do get a simular error:

```
D/Sqflite ( 6131): [Thread[Sqflite,5,main]] DELETE FROM devices WHERE id in ("?") []
E/AndroidRuntime( 6131): FATAL EXCEPTION: Sqflite
E/AndroidRuntime( 6131): Process: com.example.protectionpro, PID: 6131
E/AndroidRuntime( 6131): java.lang.IllegalArgumentException: Too many bind arguments.  1 arguments were provided but the statement needs 0 arguments.
E/AndroidRuntime( 6131): 	at android.database.sqlite.SQLiteProgram.<init>(SQLiteProgram.java:68)
E/AndroidRuntime( 6131): 	at android.database.sqlite.SQLiteStatement.<init>(SQLiteStatement.java:31)
E/AndroidRuntime( 6131): 	at android.database.sqlite.SQLiteDatabase.executeSql(SQLiteDatabase.java:1752)
E/AndroidRuntime( 6131): 	at android.database.sqlite.SQLiteDatabase.execSQL(SQLiteDatabase.java:1732)
E/AndroidRuntime( 6131): 	at com.tekartik.sqflite.SqflitePlugin.executeOrError(SqflitePlugin.java:321)
E/AndroidRuntime( 6131): 	at com.tekartik.sqflite.SqflitePlugin.update(SqflitePlugin.java:596)
E/AndroidRuntime( 6131): 	at com.tekartik.sqflite.SqflitePlugin.access$500(SqflitePlugin.java:52)
E/AndroidRuntime( 6131): 	at com.tekartik.sqflite.SqflitePlugin$5.run(SqflitePlugin.java:645)
E/AndroidRuntime( 6131): 	at android.os.Handler.handleCallback(Handler.java:790)
E/AndroidRuntime( 6131): 	at android.os.Handler.dispatchMessage(Handler.java:99)
E/AndroidRuntime( 6131): 	at android.os.Looper.loop(Looper.java:164)
E/AndroidRuntime( 6131): 	at android.os.HandlerThread.run(HandlerThread.java:65)
```

It looks like my delete query is missing a good argument here.  While it is "my fault" the code is crashing, it probably shouldn't cause an app crash that is a pain to locate. (Seriously, I was breakpointing everywhere, but I'd step over some query and for some reason it would just continue, and then I'd get a crash who knows where.  There's a lot of code involved here as I'm syncing a local DB with a remote DB). 
Adding the debug mode line definately helped a lot, so thanks for the tip!!!  It would have been much harder to find without.

Ideally, an exception should be thrown that can be caught by flutter (as in java try catches, or something, and tell's the flutter plugin to throw an error).  It probably should give you some information about the problem query as well like most database libraries.

**Comment by csga5000** [comment]
It looks like I was technically giving an argument but it's an empty string.  I suspect on iOS I didn't get an error because it's not "removing" the argument, and it may be a valid query that does nothing.  Android seems to be removing my empty argument?

Whatever the case, I fixed it by not running the query when I'm not actually removing anything.

**Comment by alextekartik** [comment]
As as sais above `?` should be without quote otherwise sqlite assume that it is the string `"?"` (hence not requiring any argument). So instead of

    DELETE FROM devices WHERE id in ("?")

it should be

    DELETE FROM devices WHERE id in (?)

**Comment by csga5000** [comment]
Right, and that's fine.  I'm just saying these errors shouldn't cause crashing exceptions and almost no information about the query (without turning on some dev function).

**Comment by alextekartik** [comment]
Ok Understood, currently I return the sqlite error as is (that is wrapped in a DatabaseException). I'll look into adding the sql query executed for convenience. But this error should indeed crash (and you can catch it) as it is a developer error (but yeah sqlite issues are a pain to debug)

**Comment by csga5000** [comment]
Okay, perhaps a crash is acceptable, though do it it being a java exception I believe it cannot be "caught" when the app is run the usual way. (That way being run to debug flutter)

**Comment by alextekartik** [comment]
Indeed I reproduce syour issue (that I did not have so far). As you said, java crashing is not acceptable so I now catch RuntimeException as you suggested. I also added a better report in the exception message to view the command executed (if any). Doing more tests, it seems I can can also get a crash on iOS that I'll need to catch. I published a 0.8.6 version that should work better for you. Let me know.
